src/form.svelte

```diff
@@ -19,6 +19,14 @@
     titlePlaceholder = tabInfo.title;
     const availableTags = await getTags().catch(() => [])
     availableTagNames = availableTags.map(tag => tag.name)
+
+    const bookmarksFound = await search(url, { limit: 1000 }).catch(() => [])
+    const matchingBookmark = bookmarksFound.filter(bookmark => bookmark.url === url)[0]
+    if (matchingBookmark) {
```

Comment:
Just filling the form fields is a bit opaque, I assume that most people would not immediately understand what's going on. I would suggest to also display a warning hint below the URL field to indicate that a bookmark for the URL already exists. A similar hint exists in the linkding [add bookmark form](https://github.com/sissbruecker/linkding/blob/master/bookmarks/templates/bookmarks/form.html#L16-L19):

![Bildschirmfoto 2021-12-09 um 22 09 11](https://user-images.githubusercontent.com/357820/145476126-2f69b1c9-c885-433e-b18c-591e2278a08b.png)

The wording would have to be changed bit. It should probably just mention that the URL is already bookmarked, and that saving this form will update the existing bookmark. Could also note about the fact that the fields were pre-filled from the existing bookmark.